src/Symfony/Component/JsonPath/Tests/JsonCrawlerTest.php

```diff
@@ -404,6 +404,228 @@ public function testAcceptsJsonPath()
         $this->assertSame('red', $result[0]['color']);
     }
 
+    /**
+     * @dataProvider provideUnicodeEscapeSequencesProvider
+     */
+    public function testUnicodeEscapeSequences(string $jsonPath, array $expected)
+    {
+        $this->assertSame($expected, self::getUnicodeDocumentCrawler()->find($jsonPath));
+    }
+
+    public static function provideUnicodeEscapeSequencesProvider(): array
+    {
+        return [
+            [
+                '$["caf\u00e9"]',
+                ['coffee'],
+            ],
+            [
+                '$["\u65e5\u672c"]',
+                ['Japan'],
+            ],
+            [
+                '$["M\u00fcller"]',
+                [],
+            ],
+
+            [
+                '$["emoji\ud83d\ude00"]',
+                ['smiley'],
+            ],
+
+            [
+                '$["tab\there"]',
+                ['with tab'],
+            ],
+            [
+                '$["new\nline"]',
+                ['with newline'],
+            ],
+            [
+                '$["quote\"here"]',
+                ['with quote'],
+            ],
+            [
+                '$["backslash\\\\here"]',
+                ['with backslash'],
```

Comment:
All those cases should be added in the single-quote provider, so that our own logic is tested (as the double quotes will use `json_decode` now)

Reply:
Added to the data provider, thanks!